### PR TITLE
Merge 2.8 into 2.9

### PR DESCRIPTION
**Pull request by manadart**
Merge from 2.8 to bring forward:
- #12629 from wallyworld/invalid-model-credentials
- #12630 from hpidcock/fix-double-upgrade
- #12623 from manadart/2.8-net-config-addr
- #12620 from wallyworld/many-actions-finish
- #12622 from manadart/2.8-relocate-net-config-source
- #12621 from manadart/2.8-relocate-net-dns
- #12608 from achilleasa/2.8-allow-bundle-to-update-existing-offers
- #12606 from jujubot/increment-to-2.8.9

Conflict resolution required for:
- api/common/network.go
- apiserver/facades/client/applicationoffers/applicationoffers.go
- apiserver/facades/client/client/status.go
- cmd/juju/application/deployer/bundlehandler.go
- go.mod
- go.sum
- scripts/win-installer/setup.iss
- snap/snapcraft.yaml
- version/version.go

### CODE REVIEW

**Review by achilleasa — APPROVED** [review]
LGTM

**Comment by manadart** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

File: api/common/network_test.go [hunk 1]
```diff
@@ -316,9 +316,9 @@ func (s *NetworkSuite) TestGetObservedNetworkConfigForOVSDevices(c *gc.C) {
 			Flags:        net.FlagUp | net.FlagBroadcast | net.FlagMulticast,
 		},
 	}
-	s.stubConfigSource.interfaceAddrs = map[string][]net.Addr{
+	s.stubConfigSource.interfaceAddrs = map[string][]network.ConfigSourceAddr{
```
Comment (manadart) [inline api/common/network_test.go:319]: Thanks for getting this.

**Pull request merged**